MultiOracle.md

```diff
@@ -0,0 +1,450 @@
+First, that `minFail` is strictly less than `maxError` so that while it is true that all differences less than `minFail` and none
+greater than `maxError` are supported, there are no guarantees made for differences in between these two parameters.
+Second, that `minFail` and `maxError` are required to be powers of `2` and not arbitrary numbers so that in reality, allowed
+variance is only expressed in terms of order of magnitude, which is a somewhat coarse measure.
+These two things together also imply that `minFail` must be at most half of `maxError` so that the gap between `minFail`
+and `maxError` where no strict guarantees can be made is of significant size.
+
+The third parameter offers a minimal remedy to these lack of guarantees.
+It is a boolean flag (`true` or `false`) called `maximizeCoverage`.
+This parameter arises from the fact that there is a decision to be made when covering the `minFail` variance to
+either cover as much as possible (up to `maxError`) or as little as possible, and this decision has no consequence
+on the number of CETs.
+That is to say that maximizing or minimizing the cases of variance between `minFail` and `maxError` require exactly
+the same number of CETs, hence this choice is left to the DLC participants.
+While this flag does improve the probabilistic guarantees which can be made for the `maxError`-`minFail` gap, it
+should be noted that these are still only probabilistic as even if `maximizeCoverage` is set to `true`, there can still
+be differences of `minFail + 1` which are unsupported in the worst case and likewise, differences of `maxError - 1`
+may still succeed even if `maximizeCoverage` is set to `false` in the worst case.
+Yet in the average case this maximizing coverage will make it much more likely that cases in the `maxError`-`minFail` gap
+will succeed and minimizing coverage will make it much more likely that these cases fail (are not supported).
+
+There exist many versions of this proposal which allow more exact guarantees, such as ones that lift either or both of the
+constraints on the variance parameters, but they all require significantly more CETs.
+This proposal can be thought of as defining the maximally coarse CET coverage algorithm which results in the fewest
+possible multiplier on the number of CETs.
+
+At a high level the algorithm works as follows (many details and some special cases are omitted).
+Designate one oracle in every t-of-t grouping to be the primary oracle and generate the set of CETs required for the DLC
+in question as would be done for a single oracle.
+This primary oracle functionally determines the execution price, meaning that the UX should be the same as if this
+primary oracle was used on its own.
+Then, for each of these CETs representing the primary oracle's attested result, label the first and last outcome covered
+by this CET as `[start, end]`  and compute the largest allowed range `[end - maxError, start + maxError]` for
```

Comment:
I don't know if it's an error or if I'm just not understanding, but aren't the end and start inverted in the second statement?

Reply:
So in the general case, `end - start < maxError` (because most CETs are small) but I agree this is very unclear here, will do better

Reply:
I'm hoping that this will be made clear by the pictures I intend to add